### PR TITLE
.travis.yml: move CI to cmake

### LINKED ISSUE #2382: Configure with CMake on Travis CI builds
All CI builds are to be converted from the generated makefiles to cmake.

### PR DESCRIPTION
**Pull request by junghans**
Fix #2382

### CODE REVIEW

**Comment by crtrott** [comment]
Does the empty line just enable Serial?

**Comment by crtrott** [comment]
Ok to test

**Comment by dalg24** [comment]
> Ok to test

Pretty sure I had whitelisted Christoph before.  Didn't I?

File: .travis.yml [hunk 1]
```diff
@@ -14,16 +14,13 @@ cache:
   - ccache
 
 env:
-  - THREADING="serial"
-  - THREADING="openmp"
-  - THREADING="pthread"
+  -
```
Comment (dalg24) [inline .travis.yml]: Why is this line empty

Comment (junghans) [reply .travis.yml]: This basically means no environment variables.

**Comment by junghans** [comment]
Do you want to:
- inject `-O3 -Werror` again?
- build with the legacy build system? (one build maybe)
- enable OPENMP for Device Parallel?
- test more or less compilers?
- run travis only for the develop and master branch? (Will only work for branches with the new ci file)
- test different CMake build types?
- turn one build in a canary build? (as a sniff test before the other builds)

**Comment by crtrott** [comment]
- inject -O3 -Werror again?
  - Yes.
- build with the legacy build system? (one build maybe)
  - One build would be good: note use the generate_makefile_bash inside the scripts testing directory
- enable OPENMP for Device Parallel?
  - Yeah for a couple builds
- test more or less compilers?
  - I think this is fine for now
- run travis only for the develop and master branch? (Will only work for branches with the new ci file)
  - were you running it for anything else?
- test different CMake build types?
  - probably should have one OpenMP debug build
- turn one build in a canary build? (as a sniff test before the other builds)
  - yeah that sounds like a good idea: like some GCC / OpenMP / Release build

Do you want to first merge this and then do all the other things?

**Comment by junghans** [comment]
Let's merge this now and then open an issue and do the remaining ones one by one, as some will need more discussion.

**Pull request merged**